mlir/lib/Dialect/Tosa/Transforms/TosaValidation.cpp

```diff
@@ -1042,6 +1042,11 @@ void TosaValidation::runOnOperation() {
       }
     }
 
+    if (failed(profileComp.checkInvalid(op))) {
```

Comment:
This check might be too far reaching for some users of the validation pass - if this is the case, we could only run it when `strictOpSpecAlignment` is specified. Happy to have a discussion on this.

Reply:
Yes, I'm leaning on making it optional. Luke's option naming is good. :) 
Do we going to have an option `--allow-invalid-op-data-type` for example, under `--tosa-validate`?

Reply:
Implemented this feature